platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java

```diff
@@ -684,10 +699,9 @@ public void setUseKey(Boolean useKey) {
         @Override
         public boolean doCredentialsMatch(AuthenticationToken token, AuthenticationInfo info) {
             if (token instanceof SAMLAuthenticationToken) {
-                SecurityToken oldToken = (SecurityToken) token.getCredentials();
-                SecurityToken newToken = (SecurityToken) info.getCredentials();
-                return oldToken.getId()
-                        .equals(newToken.getId());
+                Object oldToken = token.getCredentials();
+                Object newToken = info.getCredentials();
+                return oldToken.equals(newToken);
```

Comment:
So the thought here is we can just use Object.equals because we would have pulled oldToken from the cache?